src/app/member/components/MemberFormBuilder.tsx

```diff
@@ -0,0 +1,396 @@
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from "@/components/ui/popover";
+import { cn } from "@/lib/utils";
+import { CalendarIcon } from "lucide-react";
+import { Calendar } from "@/components/ui/calendar";
+import FormWrapper from "../../../common/FormWrapper";
+import { Label } from "@/components/ui/label";
+import { Textarea } from "@/components/ui/textarea";
+import {
+  CNIC_VALIDATOR_REGEX,
+  PHONENUMBER_VALIDATOR_REGEX,
+} from "../../../../regex";
+import { createUser } from "../actions/action";
+import { useToast } from "@/hooks/use-toast";
+
+export type UserBuilder = z.infer<typeof userBuilderSchema>;
+
+const USER_ROLES = [
+  {
+    role: "Member",
+    value: UserRole.Member,
+  },
+
+  {
+    role: "Shift Incharge",
+    value: UserRole.ShiftIncharge,
+  },
+  {
+    role: "Incharge",
+    value: UserRole.Incharge,
+  },
+];
+
+const SHIFTTIMING = [
+  {
+    time: "Shift 12:00am to 00:06am",
+    value: Shift.ShiftA,
+  },
+  {
+    time: "Shift 00:06am to 00:12pm",
+    value: Shift.ShiftB,
+  },
+  {
+    time: "Shift 00:12pm to 00:06pm",
+    value: Shift.ShiftC,
+  },
+  {
+    time: "Shift 00:06pm to 00:12am",
+    value: Shift.ShiftD,
+  },
+];
+
+export const userBuilderSchema = z.object({
+  name: z.string().min(1, {
+    message: "Name is required",
+  }),
+  phoneNumber: z
+    .string({ message: "Phone is required" })
+    .regex(PHONENUMBER_VALIDATOR_REGEX, "Phone number is not valid"),
+  date_of_birth: z.date().max(new Date(Date.now()), "under age"),
+  cnic: z.string().regex(CNIC_VALIDATOR_REGEX, "CNIC is not valid"),
+  shift: z.enum(["A", "B", "C", "D"], {
+    errorMap: () => {
+      return { message: "Select shift timing " };
+    },
+  }),
+  address: z
+    .string({ message: "Address is required" })
+    .min(10, "Address should have at least 10 characters"),
+  role: z.enum(["member", "incharge", "shift_incharge"], {
+    errorMap: () => {
+      return { message: "Select user role" };
+    },
+  }),
+  ehad_duration: z
+    .date()
+    .min(addDays(new Date(), 30), "Minmum Ehad Duration is one Month"),
+});
+
+export const MemberFormBuilder = () => {
+  const { toast } = useToast();
+  const phoneNumberMask = useMask({
+    mask: "03__-_______",
+    replacement: { _: /\d/ },
+  });
+  const cnicMask = useMask({
+    mask: "3____-_______-_",
+    replacement: { _: /\d/ },
+  });
+
+  const form = useForm<UserBuilder>({
+    resolver: zodResolver(userBuilderSchema),
+    defaultValues: {
+      name: "",
+      phoneNumber: "",
+      date_of_birth: undefined,
+      cnic: "",
+      shift: Shift.ShiftA,
+      address: "",
+      role: UserRole.Member,
+      ehad_duration: new Date(),
+    },
+    mode: "all",
+  });
+
+  const {
+    formState: { errors },
+  } = form;
+
+  const handleSubmission = async (values: UserBuilder) => {
+    const result = await createUser(values);
+    toast({
+      title: result.title,
+      description: result.message,
+    });
+    form.reset();
```

Comment:
Also, we should navigate user to list page after creation.

Reply:
fixed